### PR TITLE
Content files

### PR DESCRIPTION
**Pull request by TheJarles**
This pull request implements the basis for file uploads to the server. With the code added in this pull request you can now POST files to content/:content_id/audio and content/:content_id/resource as specified in the project backlog. 
For file uploads I decided to use the gem Carrierwave. I tried using paperclip at first, but it turns out that was deprecated in favor of ActiveStorage, which is the core rails solution for file uploads new in 5.2. After lots of tinkering I found that ActiveStorage didn't ship with the ability to customize directory paths so I finally landed on Carrierwave, which seems to do the trick for this particular application.

### CODE REVIEW

**Comment by TheJarles** [comment]
Quick update, found something that probably needs to be taken care of. Currently there is no logic in place to ensure that the file pushed to the server (bound to a content_audio or content_resource object) matches the URL it's POSTed to. If for example I were to POST a content_audio object with content_id=1 to /contents/2/audio it works just fine, even though ti shouldn't.

File: db/schema.rb [hunk 1]
```diff
@@ -10,7 +10,28 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_05_24_011519) do
+ActiveRecord::Schema.define(version: 2019_06_30_161849) do
+
+  create_table "active_storage_attachments", force: :cascade do |t|
```
Comment (jabbors) [inline db/schema.rb]: What's the purpose of these? I don't see them added by any migrations so if these tables are required for the uploading to work we need to create the tables with migrations.

We will get runtime issue in docker if the database is changed through the `schema.rb`. The way it works is the the `schema.rb` can be used to setup the database from a clean state (faster then running through the migrations one-by-one). But once the database exists, all changes must be performed through migrations.

Comment (TheJarles) [reply db/schema.rb]: Oh, good catch. That's from my tinkering with ActiveStorage, rails core file uploading solution. I guess I forgot to revert that after I decided ActiveStorage wasn't a good solution. I'll remove that.

File: db/migrate/20190630123411_add_audio_to_content_audios.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddAudioToContentAudios < ActiveRecord::Migration[5.2]
+  def change
+    add_column :content_audios, :audio, :string
```
[the remark this answers is not in the document]
Comment (TheJarles) [reply db/migrate/20190630123411_add_audio_to_content_audios.rb]: I don't see why not. I'll get right to it.

**Comment by jabbors** [comment]
>  If for example I were to POST a content_audio object with content_id=1 to /contents/2/audio it works just fine, even though ti shouldn't.

File: db/schema.rb (hunk 1 above)
Comment (jabbors) [reply db/schema.rb]: Good that it's an easy revert 😄

File: db/migrate/20190630161849_add_resource_to_content_resources.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddResourceToContentResources < ActiveRecord::Migration[5.2]
+  def change
+    add_column :content_resources, :resource, :string
```
[the remark this answers is not in the document]
Comment (TheJarles) [reply db/migrate/20190630161849_add_resource_to_content_resources.rb]: As far as I can tell, yes that's what's happening.

**Pull request merged**